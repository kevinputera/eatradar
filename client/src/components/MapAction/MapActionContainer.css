.map-action-container {
  display: flex;
  flex-direction: column;

  align-items: center;
}
