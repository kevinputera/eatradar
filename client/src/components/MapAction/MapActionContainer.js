import React from 'react';

import RefreshLocationAction from './RefreshLocationAction/RefreshLocationAction';
import ZoomAction from './ZoomAction/ZoomAction';

import './MapActionContainer.css';

function MapActionContainer(props) {
  return (
    <div className="map-action-container">
      <RefreshLocationAction refreshLocation={props.refreshLocation} />
      <ZoomAction
        handleZoomIn={props.handleZoomIn}
        handleZoomOut={props.handleZoomOut}
      />
    </div>
  );
}

export default MapActionContainer;
