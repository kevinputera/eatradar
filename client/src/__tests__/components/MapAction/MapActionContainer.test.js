import React from 'react';
import { create } from 'react-test-renderer';

import MapActionContainer from '../../../components/MapAction/MapActionContainer';

describe('tests for MapActionContainer component', () => {
  test('render must match snapshot', () => {
    const func = () => {};
    const tree = create(
      <MapActionContainer
        handleZoomIn={func}
        handleZoomOut={func}
        refreshLocation={func}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
