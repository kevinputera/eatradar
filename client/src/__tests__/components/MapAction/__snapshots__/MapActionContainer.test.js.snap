// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for MapActionContainer component render must match snapshot 1`] = `
<div
  className="map-action-container"
>
  <div
    className="refresh-location-action"
  >
    <button
      className="bp3-button bp3-large"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      style={
        Object {
          "borderRadius": "20px",
        }
      }
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-locate"
        icon="locate"
      >
        <svg
          data-icon="locate"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            locate
          </desc>
          <path
            d="M15 7h-.09A6.98 6.98 0 0 0 9 1.1V1c0-.55-.45-1-1-1S7 .45 7 1v.09A6.98 6.98 0 0 0 1.1 7H1c-.55 0-1 .45-1 1s.45 1 1 1h.1A6.969 6.969 0 0 0 7 14.91V15c0 .55.45 1 1 1s1-.45 1-1v-.09A6.98 6.98 0 0 0 14.9 9h.1c.55 0 1-.45 1-1s-.45-1-1-1zm-6.02 5.9c-.05-.5-.46-.9-.98-.9s-.93.4-.98.9A5.017 5.017 0 0 1 3.1 8.98c.5-.05.9-.46.9-.98s-.4-.93-.9-.98A5.017 5.017 0 0 1 7.02 3.1c.05.5.46.9.98.9s.93-.4.98-.9c1.97.39 3.52 1.95 3.92 3.92-.5.05-.9.46-.9.98s.4.93.9.98a5.017 5.017 0 0 1-3.92 3.92zM8 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="zoom-action"
  >
    <div
      className="zoom-control"
    >
      <button
        className="bp3-button bp3-large zoom-in-button"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "borderRadius": "20px",
          }
        }
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-plus"
          icon="plus"
        >
          <svg
            data-icon="plus"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              plus
            </desc>
            <path
              d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      className="zoom-control"
    >
      <button
        className="bp3-button bp3-large zoom-out-button"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "borderRadius": "20px",
          }
        }
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-minus"
          icon="minus"
        >
          <svg
            data-icon="minus"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              minus
            </desc>
            <path
              d="M13 7H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
